tests/scale/README.md

```diff
@@ -0,0 +1,155 @@
+# Scale testing
+## Overview
+This folder contains scale tests and related/explicit scale helpers to test OSM.
+
+Scale testing will have several use cases:
+- It will seek to qualify and document the behavior of OSM (and related control plane services) under certain loads associated to one or multiple dimensions of what a service mesh a service mesh can offer (number of services, number of policies, rate of change, etc).
+- It will help find, understand and validate soft ceilings and bottlenecks on current design and implementation.
+- It should also provide instrumentation to gate at PR level (possibly an on-demand pipeline), in order to find possible performance/scale regressions before merge.
+- It should also provide meaningful insight on how the performance and scalability of the product evolves at each milestone/release.
+
+## Design
+
+Current tests leverage the existing test framework to deploy OSM and repeat a work operation (iteration) till the validation phase of the iteration fails. How and what does the iteration do is up to the test, so implementation is free to scale any resource/s till failure.
+
+The framework provides helpers to track profiling information through the overall test and individual iterations. Some of most relevant metrics are automatically captured during the test, but more exclusive metrics for speicific tests might have to be implemented. 
+Gathering these metrics require a Prometheus instance scraping both the envoy endpoints and the K8s api servers.
+
+The current set of metrics aimed  to be automatically tracked across iteration for any set of resources specified in the test are:
+- CPU Average loads, for each iteration for each tracked resource.
+- RSS footprint and related relative increases per iteration per tracked resource.
+- Visual representation of the previous trends, provided by Grafana.
+- Control plane profiling (pprof), cpu and mem (Todo)
+- Envoy config latency trends (time to create an envoy config on OSM, latency increase per pod, test dependant) (Todo)
+- Envoy config latency apply trends (from  `SMI apply`  to  `200`  network requests) (Todo)
+
+
+## Usage examples
+Create a test using GinkGo, and import the test framework. As we do for our E2E, we try to keep its own test in a separate file.
```

Comment:
`try to keep` sounds best effort. Should we make it explicit here?

Reply:
yep, amending